msteams-platform/bots/how-to/conversations/conversation-messages.md

```diff
@@ -215,6 +223,312 @@ async def on_members_added_activity(
+
+To get an edit or undelete message event notification in a bot, you can override the following handlers:
+
+* For edit: `OnTeamsMessageEditAsync` or `OnMessageUpdateActivityAsync`
+* For undelete: `OnTeamsMessageUndeleteAsync` or `OnMessageUpdateActivityAsync`
+
+The following is an example of an edit message event notification when a sent message is edited:
+
+# [C#](#tab/csharp3)
+
+```csharp
+
+protected override async Task OnTeamsMessageEditAsync(ITurnContext<IMessageUpdateActivity> turnContext, CancellationToken cancellationToken) 
+{ 
+var replyActivity = MessageFactory.Text("message is updated"); 
+await turnContext.SendActivityAsync(replyActivity, cancellationToken); 
+} 
+
+```
+
+# [JSON](#tab/json3)
+
+```json
+
+{
+"type":"messageUpdate",
+"timestamp":"2022-10-28T17:19:39.4615413Z",
+"localTimestamp":"2022-10-28T10:19:39.4615413-07:00",
+"id":"1666977568748",
+"channelId":"msteams",
+"serviceUrl":"https://canary.botapi.skype.com/amer/",
+"from": {
+    "id":"29:1BLjP9j3_PM4mubmQZsYPx7jDyLeLf_YVA9sVPV08KMAFMjJWB_EUGveb9EVDh9TslNp9qjnzEBy3kgw01Jf1Kg",
+    "name":"Mike Wilber",
+    "aadObjectId":"520e4d1e-2108-43ee-a092-46a9507c6200"
+},
+"conversation":{
+    "conversationType":"personal",
+    "tenantId":"528dbe3f-15e0-4e37-84a1-00cc305847dd","id":"a:1pweuGJ44RkB90tiJNQ_I6g3vyuP4CYA_f-v6f0Vd-Bs3Ce85C73Ah1y8TvyjESsTHWjjgw-gnsuIuCUOWkfOCq6qaUYsk2_-fj93XXXHUMAUzhFFvTnaCU7V4WiMqRPB"
+},
+"recipient":{
+    "id":"28:0d569679-gb4j-479a-b0d8-238b6e6b1149",
+    "name":"TestBot"
+},
+"entities":[
+    {
+        "locale":"en-US",
+        "country":"US",
+        "platform":"Web",
+        "timezone":"America/Los_Angeles",
+        "type":"clientInfo"
+    }
+],
+"channelData":{
+    "eventType":"editMessage",
+    "tenant":{"id":"528dbe3f-15e0-4e37-84a1-00cc305847dd"}
+},
+"locale":"en-US",
+"localTimezone":"America/Los_Angeles"
+}  
+
+```
+
+# [Javascript](#tab/javascript3)
+
+You can use one of the following methods to get event notifications to handle the message updates using the Bot SDK:
+
+**​Event function registration**:
+
+```javascript
+
+this.onTeamsMessageEditEvent(async (context, next) => {
+  let editedMessage = context.activity.text;
+  let messageId = context.activity.id;
+  let previousMessage = this.currentMessage[messageId];
+
+  await context.sendActivity(`The previous message was "${previousMessage}". It is now: "${editedMessage}"`);
+
+  this.updateCurrentMessage(messageId, editedMessage);
+
+  next();
+})
+
+```
+
+**​Method override**:
+
+```javascript
+
+async onTeamsMessageEdit(context) {
+    let editedMessage = context.activity.text;
+    let messageId = context.activity.id;
+    let previousMessage = this.currentMessage[messageId];
+
+    await context.sendActivity(`The previous message was "${previousMessage}". It is now: "${editedMessage}"`);
+
+    this.updateCurrentMessage(messageId, editedMessage);
```

Comment:
It would be a good idea to remove line 320, 321 and 325. And replace the text in line 323 with `"The edited message is ${editedMessage}"`

Reply:
The same applies to the other method of implementation as well

Reply:
updated as suggested